### PR TITLE
On report errors from whitelist urls in sentry

### LINKED ISSUE #623: Configure sentry to ignore JS errors that don't come from our code

### PR DESCRIPTION
**Pull request by musangowope**
The intention here is to only report js sentry errors js files in the domain's `static/js`

Hypothetical demo:
https://www.loom.com/share/2b522b47f61c409294610ddce1b72fac

resolves #623

### CODE REVIEW

File: peachjam/templates/peachjam/layouts/main.html [hunk 1]
```diff
@@ -20,8 +20,12 @@
         Sentry.init({
         dsn: '{{ SENTRY_DSN_KEY }}',
         environment: '{{ SENTRY_ENVIRONMENT }}',
-        blacklistUrls: [],
-        });
+        // The list will expand as we add more sites
+        whitelistUrls: [
+          /https?:\/\/agp\.africanlii\.org/,
```
[the remark this answers is not in the document]
Comment (longhotsummer) [reply peachjam/templates/peachjam/layouts/main.html:25]: I think this is a reasonable strategy.

File: peachjam/templates/peachjam/layouts/main.html [hunk 2]
```diff
@@ -20,8 +20,8 @@
         Sentry.init({
         dsn: '{{ SENTRY_DSN_KEY }}',
         environment: '{{ SENTRY_ENVIRONMENT }}',
-        blacklistUrls: [],
-        });
+        whitelistUrls: [`${window.location.origin}/static/js/app-prod.js`]
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/layouts/main.html:23]: This needs to be a regex. In production, the app-prod.js is fingerprinted with a hash. I'd say anything that is included from static/js we should know about.

Comment (musangowope) [reply peachjam/templates/peachjam/layouts/main.html]: done

File: peachjam/templates/peachjam/layouts/main.html [hunk 3]
```diff
@@ -17,10 +17,10 @@
     <script defer src="https://browser.sentry-cdn.com/6.18.1/bundle.min.js" integrity="sha384-XOitiDSsTajhNQgwzQ7iHMlM2Q2ODCW6aw3m1QgMVrDOqsNhFao1VjB39qXNMOZW" crossorigin="anonymous"></script>
     <script>
     window.addEventListener('load', function() {
-        Sentry.init({
+      Sentry.init({
         dsn: '{{ SENTRY_DSN_KEY }}',
         environment: '{{ SENTRY_ENVIRONMENT }}',
-        blacklistUrls: [],
+        whitelistUrls: [`${window.location.hostname}/static/js/.+.js$`]
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/layouts/main.html:23]: I think this must be a regex, eg:

```js
allowUrls: [new RegExp(window.location.hostname.replace(".", "\\.") + "/static/js/.+\\.js$")]
```

Comment (musangowope) [reply peachjam/templates/peachjam/layouts/main.html]: I set up my own sentry instance and tested out this config. No events were sent to the sentry instance.

However, I managed to filter events using [beforeSend](https://github.com/laws-africa/peachjam/pull/664/files#r1022056945)

File: peachjam/templates/peachjam/layouts/main.html [hunk 4]
```diff
@@ -17,10 +17,12 @@
     <script defer src="https://browser.sentry-cdn.com/6.18.1/bundle.min.js" integrity="sha384-XOitiDSsTajhNQgwzQ7iHMlM2Q2ODCW6aw3m1QgMVrDOqsNhFao1VjB39qXNMOZW" crossorigin="anonymous"></script>
     <script>
     window.addEventListener('load', function() {
-        Sentry.init({
+      Sentry.init({
         dsn: '{{ SENTRY_DSN_KEY }}',
         environment: '{{ SENTRY_ENVIRONMENT }}',
-        blacklistUrls: [],
+        beforeSend (event){
+        return event.request.url.includes(window.location.host) ? event : null;
```
[the remark this answers is not in the document]
Comment (musangowope) [reply peachjam/templates/peachjam/layouts/main.html]: @longhotsummer let me know if you would like to chat through this if you think this doesn't handle our issue. (Perhaps there is something I have thought of)

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**